GoogleSignIn/Sources/GIDGoogleUser.m

```diff
@@ -181,6 +183,41 @@ - (OIDAuthState *) authState{
   return ((GTMAppAuthFetcherAuthorization *)self.fetcherAuthorizer).authState;
 }
 
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+- (void)addScopes:(NSArray<NSString *> *)scopes
+    presentingViewController:(UIViewController *)presentingViewController
+                  completion:(nullable void (^)(GIDUserAuth *_Nullable userAuth,
+                                                NSError *_Nullable error))completion {
+#elif TARGET_OS_OSX || TARGET_OS_MACCATALYST
+- (void)addScopes:(NSArray<NSString *> *)scopes
+    presentingWindow:(NSWindow *)presentingWindow
+          completion:(nullable void (^)(GIDUserAuth *_Nullable userAuth,
+                                        NSError *_Nullable error))completion {
+#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+
+  if (self != GIDSignIn.sharedInstance.currentUser) {
+    NSError *error = [NSError errorWithDomain:kGIDSignInErrorDomain
+                                         code:kGIDSignInErrorCodeMismatchWithCurrentUser
+                                     userInfo:nil];
+    if (completion) {
+      dispatch_async(dispatch_get_main_queue(), ^{
+        completion(nil, error);
+      });
+    }
+    return;
+  }
+  
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+  [GIDSignIn.sharedInstance addScopes:scopes
+             presentingViewController:presentingViewController
+                           completion:completion];
+#elif TARGET_OS_OSX || TARGET_OS_MACCATALYST
+  [GIDSignIn.sharedInstance addScopes:scopes
+                     presentingWindow:presentingWindow
+                           completion:completion];
+#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
```

Comment:
Similarly, apply `#if #elif #endif` only to the `presentingViewController`/`presentingWindow` line to minimize duplication.